Начальный рефакторинг

DudeWithNuke opened the pull request:
initial refactoring

si-rodkin commented on .idea/modules.xml:
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Так, это выбрасываем сразу

si-rodkin reviewed (commented):
- [x] Выбрасываем сразу файлы IDEA и java-код

si-rodkin commented on build.gradle.kts:
```diff
@@ -65,7 +65,7 @@ liquibase {
             "url" to "jdbc:postgresql://localhost:5432/suro_dev_db",
             "username" to "postgres",
-            "password" to "developer!"
+            "password" to "879587321rgk"
```
- [ ] Вот это тоже не стоит в гит кидать

si-rodkin commented on src/main/kotlin/com/example/eyeofsauron/controller/CheckPointController.kt:
```diff
@@ -0,0 +1,31 @@
+import com.example.eyeofsauron.service.CheckPointService
+import org.springframework.http.HttpStatus
+import org.springframework.web.bind.annotation.*
+
```
- [x] KDoc с пояснением что за класс и автором

si-rodkin commented on src/main/kotlin/com/example/eyeofsauron/controller/RouteBypassController.kt:
```diff
@@ -0,0 +1,31 @@
+import com.example.eyeofsauron.service.RouteBypassService
+import org.springframework.http.HttpStatus
+import org.springframework.web.bind.annotation.*
+
```
- [x] Тоже KDoc накинь

si-rodkin commented on src/main/kotlin/com/example/eyeofsauron/entity/CheckPoint.kt:
```diff
@@ -1,38 +1,34 @@
 
     val name: String,
 
-    val days: Long,
```
День зачем убрал? В схеме он есть, верни.

si-rodkin commented on src/main/kotlin/com/example/eyeofsauron/entity/CheckPoint.kt:
```diff
@@ -1,38 +1,34 @@
     val name: String,
 
-    val days: Long,
+    val readTime: LocalDateTime,
```
Это Time, а не DateTime

si-rodkin commented on src/main/kotlin/com/example/eyeofsauron/entity/CheckPoint.kt:
```diff
@@ -1,38 +1,34 @@
+    val allowanceTime: LocalDateTime,
 
-    val endTime: LocalDateTime?,
+    val bypassDate: Date,
```
Это что?

si-rodkin commented on src/main/kotlin/com/example/eyeofsauron/entity/MarkerReader.kt:
```diff
@@ -0,0 +1,20 @@
+@Entity
+data class MarkerReader(
+    @Id
+    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "markers_reader_id_seq")
```
- [x] markers_reader_id_seq?

si-rodkin commented on src/main/kotlin/com/example/eyeofsauron/entity/MarkerReader.kt:
```diff
@@ -0,0 +1,20 @@
+package com.example.eyeofsauron.entity
+
+import javax.persistence.*
+
```
- [x] KDoc

si-rodkin commented on src/main/kotlin/com/example/eyeofsauron/entity/RouteBypass.kt:
```diff
@@ -0,0 +1,21 @@
+
+import java.time.LocalDateTime
+import javax.persistence.*
+
```
- [x] KDoc

si-rodkin commented on src/main/kotlin/com/example/eyeofsauron/repository/CheckPointRepository.kt:
```diff
@@ -3,5 +3,7 @@ package com.example.eyeofsauron.repository
 import org.springframework.data.jpa.repository.JpaRepository
 import org.springframework.data.jpa.repository.JpaSpecificationExecutor
+import org.springframework.stereotype.Repository
 
```
- [x] KDoc

si-rodkin commented on src/main/kotlin/com/example/eyeofsauron/repository/CommitRepository.kt:
```diff
@@ -3,5 +3,7 @@ package com.example.eyeofsauron.repository
 import org.springframework.data.jpa.repository.JpaRepository
 import org.springframework.data.jpa.repository.JpaSpecificationExecutor
+import org.springframework.stereotype.Repository
 
```
- [x] KDoc

si-rodkin commented on src/main/kotlin/com/example/eyeofsauron/repository/MarkerReaderRepository.kt:
```diff
@@ -0,0 +1,9 @@
+import org.springframework.data.jpa.repository.JpaRepository
+import org.springframework.data.jpa.repository.JpaSpecificationExecutor
+import org.springframework.stereotype.Repository
+
```
- [x] KDoc

si-rodkin commented on src/main/kotlin/com/example/eyeofsauron/repository/RouteBypassRepository.kt:
```diff
@@ -0,0 +1,9 @@
+import org.springframework.data.jpa.repository.JpaRepository
+import org.springframework.data.jpa.repository.JpaSpecificationExecutor
+import org.springframework.stereotype.Repository
+
```
- [x] KDoc

si-rodkin commented on src/main/kotlin/com/example/eyeofsauron/repository/SecuredFacilityRepository.kt:
```diff
@@ -0,0 +1,9 @@
+import org.springframework.data.jpa.repository.JpaRepository
+import org.springframework.data.jpa.repository.JpaSpecificationExecutor
+import org.springframework.stereotype.Repository
+
```
- [x] KDoc

si-rodkin commented on src/main/kotlin/com/example/eyeofsauron/repository/UserRepository.kt:
```diff
@@ -3,5 +3,7 @@ package com.example.eyeofsauron.repository
 import org.springframework.data.jpa.repository.JpaRepository
 import org.springframework.data.jpa.repository.JpaSpecificationExecutor
+import org.springframework.stereotype.Repository
 
```
- [x] KDoc

si-rodkin commented on src/main/kotlin/com/example/eyeofsauron/service/RouteBypassService.kt:
```diff
@@ -0,0 +1,20 @@
+import com.example.eyeofsauron.entity.RouteBypass
+import com.example.eyeofsauron.repository.RouteBypassRepository
+import org.springframework.stereotype.Service
+
```
- [x] KDoc

si-rodkin committed:
Прикрутил тестовый контейнер
Добавил скрипт с тестовыми данными
Изменил тесты

DudeWithNuke committed:
Прикрутил тестовый контейнер
Добавил скрипт с тестовыми данными
Изменил тесты

DudeWithNuke committed:
Добавлено поле владельца к каждой сущности
User переименован в Employee
Добавлен PermissionService (пока только для SecuredFacility)

DudeWithNuke committed:
Прикрутил тестовый контейнер

DudeWithNuke committed:
Merge remote-tracking branch 'origin/feature/initial_refactoring' into feature/initial_refactoring

DudeWithNuke committed:
Merge remote-tracking branch 'origin/feature/initial_refactoring' into feature/initial_refactoring

DudeWithNuke committed:
Merge remote-tracking branch 'origin/feature/initial_refactoring' into feature/initial_refactoring

DudeWithNuke committed:
Добавлена проверка доступа к объектам

DudeWithNuke committed:
Добавлена проверка доступа к объектам

si-rodkin committed:
Правки

DudeWithNuke committed:
Финальные исправления

DudeWithNuke committed:
Финальные исправления
